hubblestack/extmods/module_runner/runner.py

```diff
@@ -0,0 +1,216 @@
+# -*- encoding: utf-8 -*-
+"""
+A base class for Audit/FDG module runner. 
+
+"""
+import os
+import logging
+import yaml
+from abc import ABC, abstractmethod
+from packaging import version
+
+import salt.loader
+import salt.utils
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+class Caller:
+    """
+    caller identification constant
+    """
+    AUDIT = 'AUDIT'
+    FDG = 'FDG'
+    
+class Runner(ABC):
+    """
+    Runner class for Audit/FDG modules
+    """
+    def __init__(self, salt_dict, grains_dict, opts_dict, caller):
+        super().__init__()
+        # dictionary that will load modules
+        self._hmod = {}
+        self._salt_dict = salt_dict
```

Comment:
instead of passing salt_dict, grains_dict ad opts_dict we can directly use global variables available in salt here.
(\__salt__, \__grains__, \__opts__)